File: credentials-verification-android/app/src/main/java/io/iohk/atala/prism/app/views/fragments/DeleteCredentialDialogFragment.java
package io.iohk.atala.prism.app.views.fragments;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;

import com.google.android.gms.common.SupportErrorDialogFragment;

import java.util.Optional;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.OnClick;
import io.iohk.cvp.R;
import io.iohk.atala.prism.app.utils.CredentialParse;
import io.iohk.atala.prism.app.viewmodel.CredentialsViewModel;
import io.iohk.atala.prism.app.viewmodel.dtos.CredentialDto;
import lombok.NoArgsConstructor;

import static io.iohk.atala.prism.app.utils.IntentDataConstants.CREDENTIAL_DATA_KEY;
import static io.iohk.atala.prism.app.utils.IntentDataConstants.CREDENTIAL_ID_KEY;
import static io.iohk.atala.prism.app.utils.IntentDataConstants.CREDENTIAL_TYPE_KEY;

// TODO This needs its own [ViewModel]
@NoArgsConstructor
public class DeleteCredentialDialogFragment extends CvpDialogFragment<CredentialsViewModel> {
    private static final float DELETE_ALL_CONNECTIONS_DIALOG_WIDTH = 350;
    private static final float DELETE_ALL_CONNECTIONS_DIALOG_HEIGHT = 300;
    private String credentialId;

    @BindView(R.id.credential_text)
    TextView credentialText;

    @BindView(R.id.credential_logo)
    ImageView credentialLogo;

    @Inject
    ViewModelProvider.Factory factory;

    @OnClick(R.id.cancel_button)
    void cancel() {
        this.dismiss();
        Optional.ofNullable(getTargetFragment()).ifPresent(fragment ->
                fragment.onActivityResult(getTargetRequestCode(), Activity.RESULT_CANCELED, null)
        );
    }

    @OnClick(R.id.delete_button)
    void deleteCredential() {
        getViewModel().deleteCredential(credentialId);
    }

    @Override
    public void onResume() {
        super.onResume();
        Window window = getDialog().getWindow();
        WindowManager.LayoutParams params = window.getAttributes();
        float factor = getContext().getResources().getDisplayMetrics().density;
        params.width = (int) (DELETE_ALL_CONNECTIONS_DIALOG_WIDTH * factor);
        params.height = (int) (DELETE_ALL_CONNECTIONS_DIALOG_HEIGHT * factor);
        window.setAttributes(params);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        if (getDialog() != null && getDialog().getWindow() != null) {
            getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
            getDialog().getWindow().requestFeature(Window.FEATURE_NO_TITLE);
        }
        View view = super.onCreateView(inflater, container, savedInstanceState);
        credentialId = getArguments().getString(CREDENTIAL_ID_KEY);

        CredentialDto credential = CredentialParse.parse(getArguments().getString(CREDENTIAL_TYPE_KEY), getArguments().getString(CREDENTIAL_DATA_KEY));
        credentialText.setText(CredentialUtil.getNameResource(credential.getCredentialType()));
        credentialText.setTextColor(getResources().getColor(R.color.black, null));
        credentialLogo.setImageDrawable(CredentialUtil.getLogo(credential.getCredentialType(), getContext()));

        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initObservers();
    }

    private void initObservers() {
        getViewModel().getDeleteCredentialLiveData().observe(getViewLifecycleOwner(), credentialDeleted -> {
            if (credentialDeleted) {
                this.dismiss();
                Optional.ofNullable(getTargetFragment()).ifPresent(targetFragment ->
                        targetFragment.onActivityResult(getTargetRequestCode(), Activity.RESULT_OK, null)
                );
            }
        });

        getViewModel().getShowErrorMessageLiveData().observe(getViewLifecycleOwner(), showErrorMessage -> {
            FragmentManager fm = getFragmentManager();
            SupportErrorDialogFragment.newInstance(new Dialog(getContext()))
                    .show(fm, "");
            getNavigator().showPopUp(getFragmentManager(), getResources().getString(
                    R.string.server_error_message));
        });
    }

    @Override
    protected int getViewId() {
        return R.layout.component_delete_credential_dialog;
    }

    @Override
    public CredentialsViewModel getViewModel() {
        return ViewModelProviders.of(this, factory).get(CredentialsViewModel.class);
    }
}

File: prism-android-wallet/app/src/main/java/io/iohk/atala/prism/app/ui/main/credentials/DeleteCredentialDialogFragment.kt
package io.iohk.atala.prism.app.ui.main.credentials

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.os.bundleOf
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.setFragmentResult
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.findNavController
import androidx.navigation.fragment.navArgs
import dagger.android.support.DaggerDialogFragment
import io.iohk.atala.prism.app.neo.common.EventWrapperObserver
import io.iohk.atala.prism.app.neo.common.extensions.KEY_RESULT
import io.iohk.cvp.R
import io.iohk.cvp.databinding.DialogFragmentDeleteCredentialBinding
import javax.inject.Inject

class DeleteCredentialDialogFragment : DaggerDialogFragment() {

    companion object {
        const val REQUEST_DELETE_CREDENTIAL = "REQUEST_DELETE_CREDENTIAL"
    }

    private val args: DeleteCredentialDialogFragmentArgs by navArgs()

    private lateinit var binding: DialogFragmentDeleteCredentialBinding

    @Inject
    lateinit var factory: ViewModelProvider.Factory

    private val viewModel: DeleteCredentialDialogViewModel by lazy {
        ViewModelProvider(this, factory).get(DeleteCredentialDialogViewModel::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // TODO find how to set this style within the app theme (R.style.AppTheme)
        setStyle(DialogFragment.STYLE_NO_TITLE, R.style.AlertDialogTheme)
        KEY_RESULT
        viewModel.fetchCredentialInfo(args.credentialId)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = DataBindingUtil.inflate(inflater, R.layout.dialog_fragment_delete_credential, container, false)
        binding.lifecycleOwner = this
        binding.viewModel = viewModel
        setObservers()
        return binding.root
    }

    private fun setObservers() {
        viewModel.credentialDeleted.observe(
            viewLifecycleOwner,
            EventWrapperObserver {
                if (it) {
                    findNavController().popBackStack()
                    setFragmentResult(REQUEST_DELETE_CREDENTIAL, bundleOf(KEY_RESULT to Activity.RESULT_OK))
                }
            }
        )
        viewModel.canceled.observe(
            viewLifecycleOwner,
            EventWrapperObserver {
                if (it) {
                    findNavController().popBackStack()
                    setFragmentResult(REQUEST_DELETE_CREDENTIAL, bundleOf(KEY_RESULT to Activity.RESULT_CANCELED))
                }
            }
        )
        viewModel.credential.observe(viewLifecycleOwner) {
            binding.credentialNameTextView.text = CredentialUtil.getName(it, requireContext())
            binding.credentialLogoImageView.setImageDrawable(CredentialUtil.getLogo(it.credentialType, requireContext()))
        }
    }
}
